src/init.cpp

```diff
@@ -1634,10 +1632,13 @@ bool AppInitMain()
 
     // Either install a handler to notify us when genesis activates, or set fHaveGenesis directly.
     // No locking, as this happens before any background thread is started.
```

Comment:
This comment needs to be updated at least. Preferably with an explanation why you're adding a lock here...